lightning_client.go

```diff
@@ -2680,6 +2750,150 @@ func (s *lightningClient) DescribeGraph(ctx context.Context,
 	return graph, nil
 }
 
+// SubscribeGraph allows a client to subscribe to gaph topology updates.
+func (s *lightningClient) SubscribeGraph(ctx context.Context) (
+	<-chan *GraphTopologyUpdate, <-chan error, error) {
+
+	updateStream, err := s.client.SubscribeChannelGraph(
+		s.adminMac.WithMacaroonAuth(ctx),
+		&lnrpc.GraphTopologySubscription{},
+	)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	updates := make(chan *GraphTopologyUpdate)
+	errChan := make(chan error, 1)
+
+	s.wg.Add(1)
+	go func() {
+		defer s.wg.Done()
+
+		for {
+			rpcUpdate, err := updateStream.Recv()
+			if err != nil {
+				errChan <- err
+				return
+			}
+
+			update, err := getGraphTopologyUpdate(rpcUpdate)
+			if err != nil {
+				errChan <- err
+				return
+			}
+
+			select {
+			case updates <- update:
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+
+	return updates, errChan, nil
+}
+
+// getGraphTopologyUpdate converts an lnrpc.GraphTopologyUpdate to the higher
+// level GraphTopologyUpdate.
+func getGraphTopologyUpdate(update *lnrpc.GraphTopologyUpdate) (
+	*GraphTopologyUpdate, error) {
+
+	result := &GraphTopologyUpdate{
+		NodeUpdates: make([]NodeUpdate, len(update.NodeUpdates)),
+		ChannelEdgeUpdates: make(
+			[]ChannelEdgeUpdate, len(update.ChannelUpdates),
+		),
+		ChannelCloseUpdates: make(
+			[]ChannelCloseUpdate, len(update.ClosedChans),
+		),
+	}
+
+	for i, nodeUpdate := range update.NodeUpdates {
+		identityKey, err := route.NewVertexFromStr(
+			nodeUpdate.IdentityKey,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		rawFeatureVector := &lnwire.RawFeatureVector{}
+		err = rawFeatureVector.Decode(
+			bytes.NewReader(nodeUpdate.GlobalFeatures),
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		result.NodeUpdates[i] = NodeUpdate{
+			Addresses:   nodeUpdate.Addresses,
+			IdentityKey: identityKey,
+			GlobalFeatures: *lnwire.NewFeatureVector(
+				rawFeatureVector, lnwire.Features,
+			),
+			Alias: nodeUpdate.Alias,
+			Color: nodeUpdate.Color,
+		}
+	}
+
+	for i, channelUpdate := range update.ChannelUpdates {
+		channelPoint, err := getOutPoint(
+			channelUpdate.ChanPoint.GetFundingTxidBytes(),
+			channelUpdate.ChanPoint.OutputIndex,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		advertisingNode, err := route.NewVertexFromStr(
+			channelUpdate.AdvertisingNode,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		connectingNode, err := route.NewVertexFromStr(
+			channelUpdate.ConnectingNode,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		result.ChannelEdgeUpdates[i] = ChannelEdgeUpdate{
+			ChannelID: lnwire.NewShortChanIDFromInt(
+				channelUpdate.ChanId,
+			),
+			ChannelPoint: *channelPoint,
+			Capacity:     btcutil.Amount(channelUpdate.Capacity),
+			RoutingPolicy: *getRoutingPolicy(
```

Comment:
nit: add a comment that this is always set in `lnd`'s `rpcserver.go` and therefore should never run into a nil reference error.

Reply:
Sorry, I pushed the merge button in the GitHub app for the wrong PR, will create a follow up (also making support for the new `Features` map).